library/core/src/ptr/mod.rs

```diff
@@ -56,6 +56,48 @@
 //! has size 0, i.e., even if memory is not actually touched. Consider using
 //! [`NonNull::dangling`] in such cases.
 //!
+//! ## Pointer to reference conversion
+//! When converting a pointer to a reference `&T` using `&*`,
+//! there are several rules that must be followed:
+//!
+//! * The pointer must be properly aligned.
+//!
+// some microprocessors may use address 0 for an interrupt vector.
+// users of these microprocessors must always read/write address 0 through
+// a raw pointer, not a reference.
+//! * It must be non-null.
+//!
+//! * It must be "dereferenceable" in the sense defined above.
+//!
+//! * The pointer must point to a valid instance of `T`.
+//!   This means that the created reference can only refer to
+//!   uninitialized memory through careful use of `MaybeUninit`.
+//!
+//! * You must enforce Rust's aliasing rules, since the lifetime of the
+//!   created reference is arbitrarily chosen,
+//!   and does not necessarily reflect the actual lifetime of the data.
+//!   In particular, while this reference exists,
+//!   the memory the pointer points to must
+//!   not get accessed (read or written) through any raw pointer,
+//!   except for data inside an `UnsafeCell`
+// ^ previous documentation was somewhat unclear on if modifications through
+// an UnsafeCell are safe even if they would seemingly violate the exclusivity
+// of a mut ref.
```

Comment:
Previous documentation was quite clear IMO, but note that the documentation is *different* for functions that return a mutable ref vs a shared ref. The `UnsafeCell` remark only applies to shared references. By deduplicating the docs you made the comment also apply to mutable references, which is wrong.

Reply:
Ah probably someone made a copy-paste error then, this should not ever show up in a `&mut`-returning function.